src/UI/Component/Input/Field/Factory.php

```diff
@@ -205,4 +206,60 @@ public function dependantGroup(array $inputs);
 	 * @return    \ILIAS\UI\Component\Input\Field\Checkbox
 	 */
 	public function checkbox($label, $byline = null);
+
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     A Tag Input is used to choose a subset amount of tags (techn.: array of strings) out
+	 *     of a finite list of tags. The Tag Field SHOULD be used, whenever it is not required
+	 *     or not possible to display all available options, e.g. because the amount is too high
+	 *     when the options are "all users" or "all tags.
+	 *     Besides the tags to choose from, the user can provide own tags by typing them
+	 *     into the Input (@see TagInput::withOptionsAreExtendable ).
+	 *   composition: >
+	 *     The Input is presented as a Text Input and prepended by already selected tags
+	 *     presented as texts including a Close-Button.  (e.g. [ Amsterdam X ] )
```

Comment:
I still wonder if that is correct in the technical sense. I.e.: does the Tag Field use Text Field and Close Button internally?

Reply:
Depends on the implementation, it would be possible to use it but would probably end in just passing rendered HTML from the UI-Service to JS, I'm not really fan of. But you are right, in that case we just use thing that look the same but aren't, what would be better to write down here?

Reply:
I think the crucial question here is whether you style the one alongside with the other. This to me would imply, that changes in the CSS for the Text Field would also affect the Tag Field.